blocks/rich-text/index.js

```diff
@@ -375,15 +375,33 @@ export class RichText extends Component {
 	 * Handles any case where the content of the tinyMCE instance has changed.
 	 */
 
-	onChange() {
+	onChange( options ) {
+		const { checkEquality = false } = options;
+
 		this.savedContent = this.getContent();
+
+		if ( checkEquality && isEqual( this.props.value, this.savedContent ) ) {
+			return;
+		}
+
 		this.isEmpty = isEmpty( this.savedContent, this.props.format );
 		this.props.onChange( this.savedContent );
 	}
 
-	onCreateUndoLevel() {
-		// Always ensure the content is up-to-date.
-		this.onChange();
+	onCreateUndoLevel( event ) {
+		// TinyMCE fires a `change` event when the first letter in an instance
+		// is typed. This should not create a history record in Gutenberg.
+		// https://github.com/tinymce/tinymce/blob/4.7.11/src/core/main/ts/api/UndoManager.ts#L116-L125
+		// In other cases TinyMCE won't fire a `change` with at least a previous
+		// record present, so this is a reliable check.
+		// https://github.com/tinymce/tinymce/blob/4.7.11/src/core/main/ts/api/UndoManager.ts#L272-L275
+		if ( event.lastLevel === null ) {
+			return;
+		}
+
+		// Always ensure the content is up-to-date, but avoid dispatching an
+		// action if it is (by way of input).
+		this.onChange( { checkEquality: true } );
```

Comment:
Won't this be a duplicate of the same call triggered by the `input` event handler?

Reply:
> Avoid updating the content on a change event when the content is the same. This introduces a deep equality check, but the change event does not fire often (only MCE commands and internal undo levels). I don't like this check much; the alternative is to just let it dispatch an action.

It fires on TinyMCE commands, so this is needed in those cases.